src/Microsoft.AspNet.TestHost/ClientHandler.cs

```diff
@@ -160,7 +164,28 @@ public Task<HttpResponseMessage> ResponseTask
                 get { return _responseTcs.Task; }
             }
 
-            internal void CompleteResponse()
+            public CancellationToken RequestAborted
+            {
+                get { return _requestAbortedSource.Token; }
+            }
+
+            public void AbortRequest()
+            {
+                if (!_pipelineFinished)
+                {
+                    _requestAbortedSource.Cancel();
+                }
+                _responseStream.Complete();
+            }
+
+            internal void PipelineComplete()
```

Comment:
Method names should be verbs/actions. It's not clear why you renamed most of these methods.